server/src/main/java/org/apache/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -127,11 +128,23 @@ Collection<DataSegment> retrieveUsedSegmentsForIntervals(
   );
 
   /**
-   * see {@link #retrieveUnusedSegmentsForInterval(String, Interval, Integer)}
+   * See {@link #retrieveUnusedSegmentsForInterval(String, Interval, Integer, DateTime)}.
    */
   default List<DataSegment> retrieveUnusedSegmentsForInterval(String dataSource, Interval interval)
   {
-    return retrieveUnusedSegmentsForInterval(dataSource, interval, null);
+    return retrieveUnusedSegmentsForInterval(dataSource, interval, null, null);
+  }
+
+  /**
+   * See {@link #retrieveUnusedSegmentsForInterval(String, Interval, Integer, DateTime)}.
+   */
+  default List<DataSegment> retrieveUnusedSegmentsForInterval(
```

Comment:
We should maybe get rid of some of these methods as they seem to be used only in tests.

Reply:
In general, I'm trying to be cautious with not deleting existing interface methods to ensure there are smooth upgrades and downgrades in a cluster with mixed versions. Let me think about this more and confirm.

Reply:
That makes sense, but this is internal interface used only within a single service and not while communicating across services.